File: DTC.Api/Dtos/Location/CreateLocationDto.cs
using System.ComponentModel.DataAnnotations;

namespace DTC.Api.Dtos.Location
{
    public class CreateLocationDto
    {
        [Required(ErrorMessage = "Der Name des Standorts ist ein Pflichtfeld.")]
        [StringLength(100, ErrorMessage = "Der Name des Standorts darf maximal 100 Zeichen lang sein.")]
        public string Name { get; set; } = string.Empty;

        [StringLength(250, ErrorMessage = "Die Adresse darf maximal 250 Zeichen lang sein.")]
        public string? Address { get; set; }
    }
}

File: dtc.frontend/src/dtos/location/CreateLocationDto.ts
import z from "zod";

export const createLocationDtoSchema = z.object({
  name: z
    .string({
      message: "Name ist ein Pflichtfeld.",
    })
    .min(1, "Name ist ein Pflichtfeld.")
    .max(100, "Der Name darf maximal 100 Zeichen lang sein."),

  address: z
    .string()
    .max(200, "Die Adresse darf maximal 200 Zeichen lang sein.")
    .nullable(),
});

export type CreateLocationDto = z.infer<typeof createLocationDtoSchema>;
